### ISSUE TITLE
TABULA Germany archetypes

### ISSUE THREAD

**Issue by PRemmen** [RWTH-EBC/TEASER#439]
This issue relates to #403 

### What is our Problem?

- Currently for residential buildings we only have sound archetypes and type building elements for single family dwelling following `IWU` approach (urbanrenet are based on the same building elements and similar functions)

### Why do we want to solve it?

- to also have archetypes following the accepted TABULA approach (including single, multi and terraced dwellings as well as apartment blocks)
- to make TEASER more attractive for other people
- to have a good TABULA infrastructure to enable other researchers to also include other countries

### How do we want to solve it?

- [x] Create new package for TABULA archetypes including
- [x] For each TABULA building type create one class
- [x] change calculation and estimation factors if needed 
- [x] include living area to facade area ratio into these classes
- [x] define proper type building elements in a new XML
  - chose material from MASEA data base
  - chose material thickness of each layer based on U-Values
  - also include refurbished and advanced refurbished
- [x] change API functions for TABULA buildings
- [x] change DataClass to also load TABULA information
- [x] change retrofit function in order to use information from TABULA if wanted
- [x] integrate interior walls (probably taken from iwu statistics, any better ideas?)
- [x] create table for wiki with TABULA constructions

**Comment by PRemmen** [comment]
It seems that, again, the XML bindings are kind of a mess. The reason are our faulty namespaces, from version 0.6 we will use individual namespaces for XML bindings (e.g. `http://teaser.0.6.material' ). 

I will integrate the new Materials XML from #404 in this branch as we need the new materials XML for type building elements

**Comment by PRemmen** [comment]
Okay, this has gotten quite a major change in data structure.

My suggestion is to release a new version once this branch is finished. It already includes:

- Integration of MASEA database #404 
- Integration of Tabula Germany (SFH, TH, MFH, AB), approxx. 630 new type elements for different constructions(standard, retrofit, adv retrofit)
- New Class "Door"
- a lot of new unit tests, already structured as described in #459 
- Restructuring of DataClass, now you can chose different statistics (e.g. iwu or tabula_de).

After my vacation I will work on integrating internal walls (currently they are missing completly). I will update documentation for calculation of U-Values, also provide a table for our wiki where it is clearly stated what kind of material with the corresponding thickness is used for TABULA constructions. 

Any objections to integrate the mentioned changes in a new release?

**Comment by mlauster** [comment]
No, go ahead!

**Comment by yuwash** [comment]
The teaser.project.Project.save_project function doesn't seem to be saving buildings created with TABULA. Example:

```python
from teaser.project import Project

p = Project(load_data=True)
p.name = 'example_tabula'
p.add_residential(
    method='tabula_de',
    usage='single_family_house',
    name='single_family_house_001',
    year_of_construction=2015,
    number_of_floors=5,
    height_of_floors=2.5,
    net_leased_area=250,
)
p.save_project(file_name='example_tabula', path='/tmp')
```

generates `/tmp/example_tabula.teaserXML`:

```xml
<?xml version="1.0" ?>
<project:Project version="0.6" xmlns:project="http://teaser/0.6/project"/>
```

whereas the `project:Project` node contains the building if created with IWU.

**Comment by PRemmen** [comment]
Thanks @yuwash I added a new issue for that as this requires quite some changes
